### PR TITLE
android: use EditPrefs instead of passing UpdatePrefs to start

### PR DESCRIPTION
**Pull request by oxtoacart**
This allows us to precisely set the options we need during login and avoid wiping away defaults like AllowSingleHost.

Updats tailscale/tailscale#11731

### CODE REVIEW

**Review by agottardo — APPROVED** [review]

**Comment by MulverineX** [comment]
"Updats", whoops! 😝

File: android/src/main/java/com/tailscale/ipn/ui/viewModel/IpnViewModel.kt [hunk 1]
```diff
@@ -85,14 +85,21 @@ open class IpnViewModel : ViewModel() {
 
   // Login/Logout
 
-  fun login(options: Ipn.Options = Ipn.Options(), completionHandler: (Result<Unit>) -> Unit = {}) {
-
+  fun login(
+      options: Ipn.Options = Ipn.Options(),
```
Comment (bradfitz) [inline android/src/main/java/com/tailscale/ipn/ui/viewModel/IpnViewModel.kt:89]: maybe you should just take the authKey string only? You don't have separate frontend-vs-backend log IDs on Android now, do you?

File: android/src/main/java/com/tailscale/ipn/ui/viewModel/IpnViewModel.kt [hunk 2]
```diff
@@ -102,21 +109,24 @@ open class IpnViewModel : ViewModel() {
       }
     }
 
-    Client(viewModelScope).start(options) { start ->
-      start.onFailure { completionHandler(Result.failure(it)) }.onSuccess { loginAction() }
+    val startAction = {
+      Client(viewModelScope).start(options) { start ->
+        start.onFailure { completionHandler(Result.failure(it)) }.onSuccess { loginAction() }
```
Comment (bradfitz) [inline android/src/main/java/com/tailscale/ipn/ui/viewModel/IpnViewModel.kt:114]: if this is the style, that's fine, but I personally think it's sad that onFailure and onSuccess's bodies don't line up in any way.

**Review by bradfitz — APPROVED** [review]

File: android/src/main/java/com/tailscale/ipn/ui/viewModel/IpnViewModel.kt (hunk 2 above)
Comment (oxtoacart) [reply android/src/main/java/com/tailscale/ipn/ui/viewModel/IpnViewModel.kt:114]: Yeah, that's the auto-formatter.

File: android/src/main/java/com/tailscale/ipn/ui/viewModel/IpnViewModel.kt (hunk 1 above)
Comment (oxtoacart) [reply android/src/main/java/com/tailscale/ipn/ui/viewModel/IpnViewModel.kt]: Good call.

**Comment by oxtoacart** [comment]
Thanks for the review @agottardo, @bradfitz and @MulverineX

**Pull request merged**